### ISSUE TITLE
cmake fails in target_link_libraries for co2brineflash

### ISSUE THREAD

**Issue by blattms** [OPM/opm-common#3381]
Not sure what I am doing wrong (on newly set up Ubunut 20.04 LTS with master) without any cmake options:
```
CMake Error at CMakeLists.txt:361 (target_link_libraries):
  Cannot specify link libraries for target "test_co2brine_ptflash" which is
  not built by this project.


-- Configuring incomplete, errors occurred!
````

Never so this on my Debian machine.

**Comment by blattms** [comment]
are you on ubuntu, too?

**Comment by bska** [comment]
> are you on ubuntu, too?

Yes; 20.04 LTS, GCC 9.4.x, Boost 1.71, Dune 2.7.1, CMake 3.16.3.

**Comment by blattms** [comment]
Just upgraded one of my machines and now I see that it is also an issue on Debian testing/bookworm.

**Comment by blattms** [comment]
Well, we seem to find boost unit test:
```
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required i
s "1.44.0") found components: system unit_test_framework 
```
But `Boost_UNIT_TEST_FRAMEWORK_FOUND` is still off and hence no tests are built and we cannot link to them.

**Comment by blattms** [comment]
And the problem seems to be that we have two boost searches one for components "system unit_test_framework" and another one for components "filesystem regex system unit_test_framework" where on my system regex is not found and that seems unset Boost_UNIT_TEST_FRAMEWORK_FOUND, too.

**Comment by blattms** [comment]
well there are actually 3 searches for boost between the two above there is another quiet one for just unit_test_framework
